authors/apps/article/serializers.py

```diff
@@ -18,16 +18,60 @@
 
 
 class ArticleSerializer(serializers.ModelSerializer):
+    # add the return fields
+    url = serializers.SerializerMethodField(read_only=True)
+    facebook = serializers.SerializerMethodField(read_only=True)
+    Linkedin = serializers.SerializerMethodField(read_only=True)
+    twitter = serializers.SerializerMethodField(read_only=True)
+    mail = serializers.SerializerMethodField(read_only=True)
+
     update_url = serializers.HyperlinkedIdentityField(
         view_name=NAMESPACE + ':update', lookup_field='slug')
     delete_url = serializers.HyperlinkedIdentityField(
         view_name=NAMESPACE + ':delete', lookup_field='slug')
     author = serializers.SerializerMethodField(read_only=True)
 
+    def get_url(self, obj):
+        request = self.context.get("request")
+        return obj.api_url(request=request)
+
+    def link_get(self, obj, link, args=None):
+        """get url and append the link to url that you want to share"""
+        request = self.context.get("request")
+        if args is None:
+            return link + obj.api_url(request=request)
+        if args:
+            return link + obj.api_url(request=request) + args
+
+    def get_facebook(self, obj):
+        """append facebook link."""
+        link = 'http://www.facebook.com/sharer.php?u='
+        return self.link_get(obj, link)
+
+    def get_Linkedin(self, obj):
+        """append linkedin link"""
+        link = 'http://www.linkedin.com/shareArticle?mini=true&amp;url='
```

Comment:
Shouldn't this be `https`

Reply:
> Shouldn't this be `https`

That is the shareable link I got from LinkedIn. Let me add an 's' because it will still work